code/app/src/main/java/com/example/bread/view/AddMoodEventActivity.java

```diff
@@ -0,0 +1,181 @@
+    private Spinner emotionalStateSpinner, socialSituationSpinner;
+    private EditText eventTitleEditText, reasonEditText, triggerEditText;
+    private CheckBox locationCheckbox;
+    private Button saveButton;
+    private MoodEventRepository moodEventRepository;
+    private ParticipantRepository participantRepository;
+    private FusedLocationProviderClient fusedLocationClient;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_add_mood_event);
+
+        // Tie UI elements to variables
+        emotionalStateSpinner = findViewById(R.id.emotionalStateSpinner);
+        reasonEditText = findViewById(R.id.reasonEditText);
+        socialSituationSpinner = findViewById(R.id.socialSituationSpinner);
+        saveButton = findViewById(R.id.saveButton);
+        eventTitleEditText = findViewById(R.id.eventTitleEditText);  // New field
+        triggerEditText = findViewById(R.id.triggerEditText);        // New field
+        locationCheckbox = findViewById(R.id.locationCheckbox);      // New field
+
+        // Set up repositories and location client
+        moodEventRepository = new MoodEventRepository();
+        participantRepository = new ParticipantRepository();
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(this);
+
+        // Populate emotional state spinner
+        ArrayAdapter<MoodEvent.EmotionalState> moodAdapter = new ArrayAdapter<>(this,
+                android.R.layout.simple_spinner_item, MoodEvent.EmotionalState.values());
+        moodAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        emotionalStateSpinner.setAdapter(moodAdapter);
+
+        // Populate social situation spinner
+        ArrayAdapter<MoodEvent.SocialSituation> socialAdapter = new ArrayAdapter<>(this,
+                android.R.layout.simple_spinner_item, MoodEvent.SocialSituation.values());
+        socialAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        socialSituationSpinner.setAdapter(socialAdapter);
+
+        // Save button logic
+        saveButton.setOnClickListener(v -> saveMoodEvent());
+    }
+
+    private void saveMoodEvent() {
+        // Get event title (optional)
+        String eventTitle = eventTitleEditText.getText().toString().trim();
+        if (eventTitle.isEmpty()) eventTitle = null;
+
+        // Get reason (optional)
+        String reason = reasonEditText.getText().toString().trim();
+        if (reason.isEmpty()) reason = null;
+
+        // Get trigger (optional)
+        String trigger = triggerEditText.getText().toString().trim();
+        if (trigger.isEmpty()) trigger = null;
+
+        // Get selected emotional state (required)
+        MoodEvent.EmotionalState emotionalState = (MoodEvent.EmotionalState) emotionalStateSpinner.getSelectedItem();
+        if (emotionalState == null) {
```

Comment:
Also add log statements so we know what really happened when we see an error. See `LoginPage` for example. Applies to whole file.